Source/Classes/PINRemoteImageManager.m

```diff
@@ -955,7 +975,8 @@ - (NSURLRequest *)requestWithURL:(NSURL *)url key:(NSString *)key
         request = [_requestConfigurationHandler(request) mutableCopy];
     }
     
-    [NSURLProtocol setProperty:key forKey:PINRemoteImageCacheKey inRequest:request];
+    PINRemoteImageWeakTask *container = [[PINRemoteImageWeakTask alloc] initWithTask:task];
+    [NSURLProtocol setProperty:container forKey:PINRemoteImageCacheKey inRequest:request];
```

Comment:
I think we should also update these calls because we're no longer storing the key as a property. We now store a reference to a `PINRemoteImageWeakTask *`.

So let's rename `PINRemoteImageCacheKey` to `PINRemoteImageWeakTaskKey` (and set its value to `@"PINRemoteImageWeakTaskKey"`. Also, let's rename the `container` variables to `weakTask` for complete consistency.